workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs

```diff
@@ -0,0 +1,155 @@
+using System.Linq;
+using Unity.Collections;
+using Unity.Entities;
+
+#region Diagnostic control
+
+// Disable the "variable is never assigned" for injected fields.
+#pragma warning disable 649
+
+// ReSharper disable ClassNeverInstantiated.Global
+
+#endregion
+
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    public class GameObjectWorldCommandSystem : ComponentSystem
+    {
+        private struct ReserveEntityResponseData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<Commands.WorldCommands.ReserveEntityIds.CommandResponses> Responses;
+        }
+
+        [Inject] private ReserveEntityResponseData reserveEntityResponseData;
+
+        private struct CreateEntityResponseData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<Commands.WorldCommands.CreateEntity.CommandResponses> Responses;
+        }
+
+        [Inject] private CreateEntityResponseData createEntityResponseData;
+
+        private struct DeleteEntityResponseData
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<Commands.WorldCommands.DeleteEntity.CommandResponses> Responses;
+        }
+
+        [Inject] private DeleteEntityResponseData deleteEntityResponseData;
+
+        [Inject] private GameObjectDispatcherSystem GameObjectDispatcherSystem;
+
+        private static readonly InjectableId WorldCommandResponseHandlerInjectableId =
+            new InjectableId(InjectableType.WorldCommandResponseHandler, InjectableId.NullComponentId);
+
+        protected override void OnUpdate()
+        {
+            for (var i = 0; i < reserveEntityResponseData.Length; ++i)
+            {
+                if (reserveEntityResponseData.Responses[i].Responses.Count == 0)
```

Comment:
Unsure, shall I remove these checks

Reply:
Will check but I think we guarantee if it's there it has something in it.

Reply:
It can strictly happen if a system has deliberately removed all the responses. However in that case nothing would break here. So this is just an efficiency call. In which case it's probably more efficient not to make the check.

Reply:
Roger, removing the check